### ISSUE TITLE
Hunter Auto Shot damage formula changing based on number of % damage modifiers applied

### ISSUE THREAD

**Issue by watchyoursixx** [watchyoursixx/hunter-tbc#40]
While testing damage formulas I noticed something very odd happening with auto shot.

Testing with Fine Light Crossbow (29 damage, 2.7 speed) and 22 DPS ammo at 1145 AP gives me 309-310 damage range. This matches the tooltip's 309-310 damage range. The formula is:

```22*2.7 + 29 + 1145*2.7/14```

Attacking a lvl 14 borer in RFC, I see the damage goes from 309-310 to 284-285. So the mitigation is about 8%~ reduction (91.973%).

Doing this same test with Focused Fire (2% damage bonus) I get the result of tooltip showing 314-315 or (308-309)* 102%. 
If I calculate the above formula with 1.02 * the whole thing I get this:

```(22*2.7 + 29 + 1145*2.7/14)*1.02 = 315 - 316 range```

However from the tooltip you can see it's actually slightly lower, so I adjust the formula to not count the ammo damage bonus as multiplied like this:

```22*2.7 + (29 + 1145*2.7/14)*1.02 = 314 - 315 range```

Now, if I attack this same lvl 14 borer, I get 290 always. Which seems to follow along the lines of the latter formula. Notice that my damage range here also matches the range on the tooltip (for some reason it shows 308-309)

I see this same thing with Ranged Weapon spec, or ferocious inspiration. Basically up to 2 percent modified bonuses being multiplied together. As an example:

With ammo damage multiplied:
```(22*2.7 + 29 + 1145*2.7/14)*1.02*1.05 = 331-332 range```

Without ammo multiplied:
```22*2.7 + (29 + 1145*2.7/14)*1.02*1.05 =  326-327 range```

Attacking the lvl 14 borer gives 299-300, which follows the latter's formula and ammo damage not being applied to the multipliers. However, if I add in a third, say RWS and Ferocious inspiration together, or slaying % bonus, I get the following result with say these multipliers: Focused Fire (2%), Ranged Weapon Spec (5%), and Slaying (3%)

With ammo damage multiplied:
```(22*2.7 + 29 + 1145*2.7/14)*1.02*1.05*1.03 = 341-342 range```

Without ammo multiplied:
```22*2.7 + (29 + 1145*2.7/14)*1.02*1.05*1.03 = 334-335 range```

The tooltip matches the latter, but when I actually damage a mob I end up with 313-314 which matches the formula for the first formula. If the damage range actually matched 334-335 like the tooltip suggests, and the test with 1 or 2 modifiers suggests, I'd have 307-308 damage instead. 

So basically, when you have 3 or more multipliers for damage on Auto Shot, the formula literally is changing how it's calculated even though the tooltip still matches the original formula with only 1 or 2 modifiers. Maybe I missed something in there and someone can double check, but this is somewhat significant difference when you have higher damage abilities and ranges.

TL;DR 3 or more damage multipliers makes Auto Shot behave differently. All other damaging shots behave the same way regardless of % modifiers (they have ammo damage multiplied always)

**Comment by Aggrend** [comment]
We've seen other issues with damage multipliers applying incorrectly recently, so this might be a similar issue or by-product of a larger global issue. It's on the radar now though. Thanks!

**Comment by watchyoursixx** [comment]
This appears to behaving consistently now (Its hard to test with more than 3, but doing the same test above I get different results now)

It follows the "without ammo multiplied" formula
Without ammo multiplied:
22*2.7 + (29 + 586*2.7/14)*1.02*1.05*1.03 = 191-192 range
